[BUG] fix `TestHandleAccessories`

### Type of bug

Code - general bugs related to how the game runs and responds to you.

### Description

6 of the tests in `TestHandleAccessories` fail, 5 of them with `AttributeErrors` on the mock `PeltClass` and one of them with an AssertionError.

Note: these tests do not seem be run on GitHub actions.

### Grade

Non-vital - This bug can easily be ignored and does not have a large impact on play

### Steps to reproduce the behavior

I was working on one of my own branches when I stumbled upon these unit tests failing. 
Just run unit tests (I used `python3 -m unittest` on latest development to confirm).

### Patrol ID + Outcome Print (if applicable)

N/A

### Commit # or Game Version Number
https://github.com/ClanGenOfficial/clangen/commit/0d225b0cebfe49b5915d645e83a92087ba30a2d4

### Additional context

Here are the errors I get: 

```py

Error
Traceback (most recent call last):
  File "/Users/rusty/coding/clangen/tests/test_handle_short_events.py", line 66, in test_cat_gets_random_collar_accessory
    self.test.handle_accessories(pelts=self.pelt_class)
  File "/Users/rusty/coding/clangen/scripts/events_module/short/handle_short_events.py", line 404, in handle_accessories
    accessory_groups = [pelts.collars, pelts.head_accessories, pelts.tail_accessories, pelts.body_accessories]
                                       ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'PeltClass' has no attribute 'head_accessories'. Did you mean: 'wild_accessories'?


Error
Traceback (most recent call last):
  File "/Users/rusty/coding/clangen/tests/test_handle_short_events.py", line 60, in test_cat_gets_random_plant_accessory
    self.test.handle_accessories(pelts=self.pelt_class)
  File "/Users/rusty/coding/clangen/scripts/events_module/short/handle_short_events.py", line 404, in handle_accessories
    accessory_groups = [pelts.collars, pelts.head_accessories, pelts.tail_accessories, pelts.body_accessories]
                                       ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'PeltClass' has no attribute 'head_accessories'. Did you mean: 'wild_accessories'?


Error
Traceback (most recent call last):
  File "/Users/rusty/coding/clangen/tests/test_handle_short_events.py", line 54, in test_cat_gets_random_wild_accessory
    self.test.handle_accessories(pelts=self.pelt_class)
  File "/Users/rusty/coding/clangen/scripts/events_module/short/handle_short_events.py", line 404, in handle_accessories
    accessory_groups = [pelts.collars, pelts.head_accessories, pelts.tail_accessories, pelts.body_accessories]
                                       ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'PeltClass' has no attribute 'head_accessories'. Did you mean: 'wild_accessories'?


Failure
Traceback (most recent call last):
  File "/Users/rusty/coding/clangen/tests/test_handle_short_events.py", line 49, in test_cat_gets_test_accessory
    self.assertIs(self.test.main_cat.pelt.accessory, "TEST")
AssertionError: ['TEST'] is not 'TEST'


Error
Traceback (most recent call last):
  File "/Users/rusty/coding/clangen/tests/test_handle_short_events.py", line 80, in test_halftail_cats_do_not_get_tail_accessories
    self.test.handle_accessories(pelts=self.pelt_class)
  File "/Users/rusty/coding/clangen/scripts/events_module/short/handle_short_events.py", line 404, in handle_accessories
    accessory_groups = [pelts.collars, pelts.head_accessories, pelts.tail_accessories, pelts.body_accessories]
                                       ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'PeltClass' has no attribute 'head_accessories'. Did you mean: 'wild_accessories'?


Error
Traceback (most recent call last):
  File "/Users/rusty/coding/clangen/tests/test_handle_short_events.py", line 73, in test_notail_cats_do_not_get_tail_accessories
    self.test.handle_accessories(pelts=self.pelt_class)
  File "/Users/rusty/coding/clangen/scripts/events_module/short/handle_short_events.py", line 404, in handle_accessories
    accessory_groups = [pelts.collars, pelts.head_accessories, pelts.tail_accessories, pelts.body_accessories]
                                       ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'PeltClass' has no attribute 'head_accessories'. Did you mean: 'wild_accessories'?
```

### Screenshots

_No response_

--- a/scripts/cat/cats.py
+++ b/scripts/cat/cats.py
@@ -2032,15 +2032,15 @@ class Cat:
             self.pelt.accessory = [
                 acc for acc in self.pelt.accessory
                 if acc not in (
-                "RED FEATHERS",
-                "BLUE FEATHERS",
-                "JAY FEATHERS",
-                "GULL FEATHERS",
-                "SPARROW FEATHERS",
-                "CLOVER",
-                "DAISY",
-            )
-        ]
+                    "RED FEATHERS",
+                    "BLUE FEATHERS",
+                    "JAY FEATHERS",
+                    "GULL FEATHERS",
+                    "SPARROW FEATHERS",
+                    "CLOVER",
+                    "DAISY",
+                )
+            ]
 
         condition = PERMANENT[name]
         new_condition = False

--- a/scripts/cat/cats.py
+++ b/scripts/cat/cats.py
@@ -2028,26 +2028,19 @@ class Cat:
             return
 
         # remove accessories if need be
-        if "NOTAIL" in self.pelt.scars and self.pelt.accessory in [
-            "RED FEATHERS",
-            "BLUE FEATHERS",
-            "JAY FEATHERS",
-            "GULL FEATHERS",
-            "SPARROW FEATHERS",
-            "CLOVER",
-            "DAISY",
-        ]:
-            self.pelt.accessory = None
-        if "HALFTAIL" in self.pelt.scars and self.pelt.accessory in [
-            "RED FEATHERS",
-            "BLUE FEATHERS",
-            "JAY FEATHERS",
-            "GULL FEATHERS",
-            "SPARROW FEATHERS",
-            "CLOVER",
-            "DAISY",
-        ]:
-            self.pelt.accessory = None
+        if "NOTAIL" in self.pelt.scars or "HALFTAIL" in self.pelt.scars:
+            self.pelt.accessory = [
+                acc for acc in self.pelt.accessory
+                if acc not in (
+                "RED FEATHERS",
+                "BLUE FEATHERS",
+                "JAY FEATHERS",
+                "GULL FEATHERS",
+                "SPARROW FEATHERS",
+                "CLOVER",
+                "DAISY",
+            )
+        ]
 
         condition = PERMANENT[name]
         new_condition = False

--- a/scripts/cat/pelts.py
+++ b/scripts/cat/pelts.py
@@ -846,6 +846,8 @@ class Pelt:
 
         if isinstance(self.accessory, str):
             self.accessory = [self.accessory]
+        elif self.accessory is None:
+            self.accessory = []
 
 
     def init_eyes(self, parents):

--- a/scripts/cat/pelts.py
+++ b/scripts/cat/pelts.py
@@ -844,10 +844,10 @@ class Pelt:
         elif self.pattern == "MINIMAL4":
             self.pattern = "MINIMALFOUR"
 
-        if isinstance(self.accessory, str):
-            self.accessory = [self.accessory]
-        elif self.accessory is None:
+        if self.accessory is None:
             self.accessory = []
+        elif isinstance(self.accessory, str):
+            self.accessory = [self.accessory]
 
 
     def init_eyes(self, parents):

--- a/scripts/cat/pelts.py
+++ b/scripts/cat/pelts.py
@@ -1233,9 +1233,11 @@ class Pelt:
             acc_display_choice = random.randint(0, 100)
 
         if acc_display_choice == 1:
-            self.accessory = [choice(
-                [choice(Pelt.plant_accessories), choice(Pelt.wild_accessories)]
-            )]
+            self.accessory = [
+                choice(
+                    [choice(Pelt.plant_accessories), choice(Pelt.wild_accessories)]
+                )
+            ]
         else:
             self.accessory = []
 

--- a/scripts/events_module/relationship/pregnancy_events.py
+++ b/scripts/events_module/relationship/pregnancy_events.py
@@ -896,7 +896,7 @@ class Pregnancy_Events:
                     kit.relationships[the_cat.ID] = Relationship(kit, the_cat)
 
             #### REMOVE ACCESSORY ######
-            kit.pelt.accessory = None
+            kit.pelt.accessory = []
             clan.add_cat(kit)
 
             #### GIVE HISTORY ######

--- a/scripts/events_module/short/scar_events.py
+++ b/scripts/events_module/short/scar_events.py
@@ -213,16 +213,18 @@ class Scar_Events:
 
             specialty = random.choice(scar_pool)
             if specialty in ["NOTAIL", "HALFTAIL"]:
-                if cat.pelt.accessory in [
-                    "RED FEATHERS",
-                    "BLUE FEATHERS",
-                    "JAY FEATHERS",
-                    "GULL FEATHERS",
-                    "SPARROW FEATHERS",
-                    "CLOVER",
-                    "DAISY",
-                ]:
-                    cat.pelt.accessory = None
+                cat.pelt.accessory = [
+                    acc for acc in cat.pelt.accessory
+                    if acc not in (
+                        "RED FEATHERS",
+                        "BLUE FEATHERS",
+                        "JAY FEATHERS",
+                        "GULL FEATHERS",
+                        "SPARROW FEATHERS",
+                        "CLOVER",
+                        "DAISY"
+                    )
+                ]
 
             # combining left/right variations into the both version
             if "NOLEFTEAR" in cat.pelt.scars and specialty == "NORIGHTEAR":

--- a/scripts/game_structure/load_cat.py
+++ b/scripts/game_structure/load_cat.py
@@ -410,7 +410,7 @@ def csv_load(all_cats):
                 ] = "10There was an error loading cat # " + str(attr[0])
                 the_cat.skill = attr[25]
                 if len(attr) > 28:
-                    the_cat.pelt.accessory = attr[28]
+                    the_cat.pelt.accessory = [attr[28]]
                 if len(attr) > 29:
                     the_cat.specialty2 = attr[29]
                 else:

--- a/scripts/screens/ProfileScreen.py
+++ b/scripts/screens/ProfileScreen.py
@@ -47,7 +47,7 @@ from ..ui.icon import Icon
 def accessory_display_name(cat):
     accessory = cat.pelt.accessory
 
-    if accessory is None:
+    if not accessory:
         return ""
 
     return i18n.t(f"cat.accessories.{accessory}", count=0)
@@ -331,7 +331,7 @@ class ProfileScreen(Screens):
                 self.build_profile()
                 self.update_disabled_buttons_and_text()
             elif event.ui_element == self.destroy_accessory_button:
-                self.the_cat.pelt.accessory = None
+                self.the_cat.pelt.accessory = []
                 self.clear_profile()
                 self.build_profile()
                 self.update_disabled_buttons_and_text()

--- a/scripts/screens/ProfileScreen.py
+++ b/scripts/screens/ProfileScreen.py
@@ -42,18 +42,6 @@ from ..ui.icon import Icon
 
 
 # ---------------------------------------------------------------------------- #
-#             change how accessory info displays on cat profiles               #
-# ---------------------------------------------------------------------------- #
-def accessory_display_name(cat):
-    accessory = cat.pelt.accessory
-
-    if not accessory:
-        return ""
-
-    return ', '.join(i18n.t(f"cat.accessories.{acc}", count=0) for acc in accessory)
-
-
-# ---------------------------------------------------------------------------- #
 #               assigns backstory blurbs to the backstory                      #
 # ---------------------------------------------------------------------------- #
 def bs_blurb_text(cat):

--- a/scripts/screens/ProfileScreen.py
+++ b/scripts/screens/ProfileScreen.py
@@ -50,7 +50,7 @@ def accessory_display_name(cat):
     if not accessory:
         return ""
 
-    return i18n.t(f"cat.accessories.{accessory}", count=0)
+    return ', '.join(i18n.t(f"cat.accessories.{acc}", count=0) for acc in accessory)
 
 
 # ---------------------------------------------------------------------------- #

--- a/tests/test_handle_short_events.py
+++ b/tests/test_handle_short_events.py
@@ -51,7 +51,7 @@ class TestHandleAccessories(unittest.TestCase):
         self.test.chosen_event.new_accessory = ["TEST"]
 
         self.test.handle_accessories()
-        self.assertIs(self.test.main_cat.pelt.accessory, ["TEST"])
+        self.assertEqual(self.test.main_cat.pelt.accessory, ["TEST"])
 
     def test_cat_gets_random_wild_accessory(self):
         self.test.chosen_event.new_accessory = ["WILD"]
@@ -76,14 +76,14 @@ class TestHandleAccessories(unittest.TestCase):
         self.test.main_cat.pelt.scars = "NOTAIL"
 
         self.test.handle_accessories(pelts=self.pelt_class)
-        self.assertIsNone(self.test.main_cat.pelt.accessory)
+        self.assertFalse(self.test.main_cat.pelt.accessory)
 
     def test_halftail_cats_do_not_get_tail_accessories(self):
         self.test.chosen_event.new_accessory = self.pelts.tail_accessories
         self.test.main_cat.pelt.scars = "HALFTAIL"
 
         self.test.handle_accessories(pelts=self.pelt_class)
-        self.assertIsNone(self.test.main_cat.pelt.accessory)
+        self.assertFalse(self.test.main_cat.pelt.accessory)
 
 
 class TestHandleDeath(unittest.TestCase):

--- a/tests/test_handle_short_events.py
+++ b/tests/test_handle_short_events.py
@@ -39,7 +39,9 @@ class TestHandleAccessories(unittest.TestCase):
         self.pelts = self.pelt_class()
 
     def assert_intersection(self, a, b):
-        self.assertTrue(bool(set(a) & set(b)))
+        """assert that the intersection of iterables a and b is non-empty"""
+        
+        self.assertTrue(set(a) & set(b))
 
     def test_misc_appended_to_types(self):
         self.test.types = []

--- a/tests/test_handle_short_events.py
+++ b/tests/test_handle_short_events.py
@@ -38,6 +38,9 @@ class TestHandleAccessories(unittest.TestCase):
         self.test.main_cat = Cat()
         self.pelts = self.pelt_class()
 
+    def assert_intersection(self, a, b):
+        self.assertTrue(bool(set(a) & set(b)))
+
     def test_misc_appended_to_types(self):
         self.test.types = []
 
@@ -54,19 +57,19 @@ class TestHandleAccessories(unittest.TestCase):
         self.test.chosen_event.new_accessory = ["WILD"]
 
         self.test.handle_accessories(pelts=self.pelt_class)
-        self.assertIn(self.test.main_cat.pelt.accessory, self.pelts.wild_accessories)
+        self.assert_intersection(self.test.main_cat.pelt.accessory, self.pelts.wild_accessories)
 
     def test_cat_gets_random_plant_accessory(self):
         self.test.chosen_event.new_accessory = ["PLANT"]
 
         self.test.handle_accessories(pelts=self.pelt_class)
-        self.assertIn(self.test.main_cat.pelt.accessory, self.pelts.plant_accessories)
+        self.assert_intersection(self.test.main_cat.pelt.accessory, self.pelts.plant_accessories)
 
     def test_cat_gets_random_collar_accessory(self):
         self.test.chosen_event.new_accessory = ["COLLAR"]
 
         self.test.handle_accessories(pelts=self.pelt_class)
-        self.assertIn(self.test.main_cat.pelt.accessory, self.pelts.collars)
+        self.assert_intersection(self.test.main_cat.pelt.accessory, self.pelts.collars)
 
     def test_notail_cats_do_not_get_tail_accessories(self):
         self.test.chosen_event.new_accessory = self.pelts.tail_accessories

--- a/tests/test_handle_short_events.py
+++ b/tests/test_handle_short_events.py
@@ -27,7 +27,9 @@ class TestHandleAccessories(unittest.TestCase):
             wild_accessories=["WILD1", "WILD2"],
             plant_accessories=["PLANT1", "PLANT2"],
             collars=["COLLAR1", "COLLAR2"],
-            tail_accessories=["TAIL1", "TAIL2"]
+            head_accessories=["HEAD1", "HEAD2"],
+            body_accessories=["BODY1", "BODY2"],
+            tail_accessories=["TAIL1", "TAIL2"],
         ))
 
     def setUp(self):
@@ -46,7 +48,7 @@ class TestHandleAccessories(unittest.TestCase):
         self.test.chosen_event.new_accessory = ["TEST"]
 
         self.test.handle_accessories()
-        self.assertIs(self.test.main_cat.pelt.accessory, "TEST")
+        self.assertIs(self.test.main_cat.pelt.accessory, ["TEST"])
 
     def test_cat_gets_random_wild_accessory(self):
         self.test.chosen_event.new_accessory = ["WILD"]

